the-collab-lab/tcl-12-smart-shopping-list
# #5: Add functionality for user to join an existing shopping list

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

- Created home page and separate button for joining an existing list
- Created form for user to enter an existing token to join a list
- Added alerts to either create a new token or enter correct token if user enters invalid token or enters nothing


## Related Issue

Closes #5 

## Acceptance Criteria

- When the user does not already have a token in localStorage, on the onboarding/home screen, a simple form is displayed that allows the user to enter a token
- Entering the token and hitting submit saves the token to localStorage, effectively giving them joint control of the list
- On submit, show an error if the token does not exist
- If they get an error message, allow them to try again or create a new list

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

![95946453-2d488800-0db2-11eb-9afb-08a4d3090551](https://user-images.githubusercontent.com/61370161/96740557-9158e680-1386-11eb-9d36-6988b685363a.gif)


### After

![existingToken](https://user-images.githubusercontent.com/61370161/96740498-82723400-1386-11eb-96f7-e1dd6a852273.gif)


## Testing Steps / QA Criteria

- Switch to our branch (`git checkout gj-eb-join-shopping-list`), then run `git pull origin gj-eb-join-shopping-list`. Start server. 
- If there is already a token saved in localStorage, delete token by: 
    Chrome: Inspect > Application > Local Storage > Right click to delete token
    Safari: Inspect > Storage > Local Storage
    Firefox: Inspect > Storage > Local Storage
- Enter nothing or an invalid token (for ex: "hello there") then click "Join an Existing Token" button. An alert should appear stating a valid token must be inputted. 
- Enter an existing token (for ex: "rev vast petty") then click "Join an Existing Token" button. You should be redirected to the list view for that token. 
- Create a new token should work the same.
